### DISCUSSION TITLE
C# 11 required Keyword with [ObservableProperty]?

### DISCUSSION THREAD

**Discussion by cmiles** [CommunityToolkit/dotnet#634] [Q&A] [answered]
Is there any current way to use the C# 11 required keyword with [ObservableProperty]? Trying something in the style of:

[ObservableProperty] private required string _writtenFile;

Results in an error because the private modifier is less visible the the containing type (in my case the class is public).

Thanks for any information!

**Comment by Sergio0694** [comment, marked as answer]
That's not supported, as `required` is only valid on properties. It will be supported by #555 🙂
